### PR TITLE
feat: add GitHub action pr title check b

### PR DESCRIPTION
**Pull request by seedy**
# Description

Need: Secure commit messages to ensure semantic-release workflow will work.

Adds a Github action to check PR title conforms with [conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/)

The action logs nothing if the check passes, else logs errors.

PR split from https://github.com/traefik/faency/pull/288

# Action: 
- https://github.com/marketplace/actions/semantic-pull-request

Because we work on a fork-based repo, we must trigger the action on `pull_request_target`, because the GH action runs `octokit` calls with a token. More info on the repo:
- https://github.com/amannn/action-semantic-pull-request#event-triggers
- https://github.com/amannn/action-semantic-pull-request/issues/139

## Alternatives

GH actions:
- https://github.com/marketplace/actions/conventional-commits-pr seems less efficient
- https://github.com/google-github-actions/release-please-action too big for the need

Github bot/app instead of GH action:
- https://github.com/zeke/semantic-pull-requests a bot would be useful if we planned this for multiple projects

# How to test

The GH action will be able to run only once we merge this PR. To test, please have a look at:
- https://github.com/matthieuh/test-semantic-release

In this repo, I added the GH action.

To try it, simply create a PR, or try this dedicated open PR:
- https://github.com/matthieuh/test-semantic-release/pull/9

### CODE REVIEW

**Comment by ldez** [comment]
Have you tried https://github.com/marketplace/actions/conventional-commits-pr?

**Comment by seedy** [comment]
> Have you tried https://github.com/marketplace/actions/conventional-commits-pr? or https://github.com/marketplace/actions/semantic-pull-request?

@ldez Will try next week! I was wondering if it was better to create our own github action, seen the simplicity of the need.

I need to look into the actions your proposed.

**Comment by seedy** [comment]
Filed an issue on dep repo: https://github.com/conventional-changelog/commitlint/issues/3049

**Review by gndz07 — APPROVED** [review]
Tested, looks like it works as expected 👏

**Review by ldez — APPROVED** [review]

File: .github/workflows/pr_title.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 'lts/*'
+
+      - name: Install dependencies
+        run: npm ci
```
Comment (paulocfjunior) [inline .github/workflows/pr_title.yml:22]: >  ❓Question

Do we need to install deps to do the title check?

**Review by paulocfjunior — APPROVED** [review]

File: .github/workflows/pr_title.yml (hunk 1 above)
Comment (seedy) [reply .github/workflows/pr_title.yml:22]: No longer , good call!

**Pull request merged**